flow-preferences/src/main/java/com/fredporciuncula/flow/preferences/FlowSharedPreferences.kt

```diff
@@ -4,70 +4,154 @@ import android.content.SharedPreferences
+   */
   @JvmOverloads
   fun getNullableStringSet(key: String, defaultValue: Set<String>? = null): Preference<Set<String>?> =
-    NullableStringSetPreference(key, defaultValue, keyFlow, sharedPreferences, coroutineContext)
-
+    NullableStringSetPreference(key, defaultValue, keyFlow, sharedPreferences, Dispatchers.IO + coroutineContext)
+
+  /**
+   * Creates a [Preference] wrapper for [Set] of nullable [String] by a given key from [SharedPreferences]. By default, all operations will be executed on [Dispatchers.IO]
+   * @param key key for a [SharedPreferences] storage value
+   * @param defaultValue will be returned if value by given key is not found
+   * @return [Preference] wrapper of [Set] of nullable [String]
+   */
   @JvmOverloads
   fun getStringSetOfNullables(key: String, defaultValue: Set<String?> = emptySet()): Preference<Set<String?>> =
-    StringSetOfNullablesPreference(key, defaultValue, keyFlow, sharedPreferences, coroutineContext)
-
+    StringSetOfNullablesPreference(key, defaultValue, keyFlow, sharedPreferences, Dispatchers.IO + coroutineContext)
+
+  /**
+   * Creates a [Preference] wrapper for nullable [Set] of nullable [String] by a given key from [SharedPreferences]. By default, all operations will be executed on [Dispatchers.IO]
+   * @param key key for a [SharedPreferences] storage value
+   * @param defaultValue will be returned if value by given key is not found
+   * @return [Preference] wrapper of nullable [Set] of nullable [String]
+   */
   @JvmOverloads
   fun getNullableStringSetOfNullables(key: String, defaultValue: Set<String?>? = null): Preference<Set<String?>?> =
-    NullableStringSetOfNullablesPreference(key, defaultValue, keyFlow, sharedPreferences, coroutineContext)
-
+    NullableStringSetOfNullablesPreference(key, defaultValue, keyFlow, sharedPreferences, Dispatchers.IO + coroutineContext)
+
+  /**
+   * Creates a [Preference] wrapper for [Any] by a given key from [SharedPreferences]. By default, all operations will be executed on [Dispatchers.IO]
+   * @param key key for a [SharedPreferences] storage value
+   * @param defaultValue will be returned if value by given key is not found
+   * @return [Preference] wrapper of [Any] subtype
+   */
   fun <T : Any> getObject(
     key: String,
     serializer: Serializer<T>,
     defaultValue: T
   ): Preference<T> =
-    ObjectPreference(key, serializer, defaultValue, keyFlow, sharedPreferences, coroutineContext)
-
+    ObjectPreference(key, serializer, defaultValue, keyFlow, sharedPreferences, Dispatchers.IO + coroutineContext)
+
+  /**
+   * Creates a [Preference] wrapper for nullable [Any] by a given key from [SharedPreferences]. By default, all operations will be executed on [Dispatchers.IO]
+   * @param key key for a [SharedPreferences] storage value
+   * @param defaultValue will be returned if value by given key is not found
+   * @return [Preference] wrapper of nullable [Any] subtype
+   */
   fun <T> getNullableObject(
```

Comment:
💡 I guess this one should also mention the serializer:

>Creates a [Preference] of type [T] with the given `key` and `defaultValue`. A [Serializer] of the same type must be passed so the value can be stored as a [String].

Reply:
I think the first paragraph already explains everything and we could just drop the rest!